src/Analyzers/CSharp/CodeFixes/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs

```diff
@@ -124,16 +133,35 @@ internal static class MakeLocalFunctionStaticCodeFixHelper
                     {
                         var referenceSpan = location.Location.SourceSpan;
                         if (!localFunction.FullSpan.Contains(referenceSpan))
-                        {
                             continue;
-                        }
 
                         var referenceNode = root.FindNode(referenceSpan);
                         if (referenceNode is IdentifierNameSyntax identifierNode)
                         {
                             syntaxEditor.ReplaceNode(
                                 identifierNode,
-                                (node, generator) => SyntaxFactory.IdentifierName(parameter.Name.ToIdentifierToken()).WithTriviaFrom(node));
+                                (node, generator) => IdentifierName(parameter.Name.ToIdentifierToken()).WithTriviaFrom(node));
+                        }
+                    }
+                }
+            }
+
+            // If we captured 'this', then we have to go through and rewrite all usages of it to @this.  Note that
+            // 'this' may be used explicitly or implicitly.
+            if (thisParameter != null)
+            {
+                var localFunctionBodyOperation = semanticModel.GetOperation(localFunction.Body ?? (SyntaxNode)localFunction.ExpressionBody!.Expression, cancellationToken);
+                foreach (var descendent in localFunctionBodyOperation.DescendantsAndSelf())
+                {
+                    if (descendent is IInstanceReferenceOperation { ReferenceKind: InstanceReferenceKind.ContainingTypeInstance } instanceReference)
```

Comment:
❔ Do we have any tests covering cases where the value is a different instance of the same type?

```
class Example
{
  void Method(Example e)
  {
    void LocalFunction(Example e2)
    {
      Method();
      e2.Method();
    }
  }
}
```

Reply:
no.  but happy to add.  in that case it would not be an IInstanceRefOperation.  But more coverage is good :)

Reply:
added.